swri_roscpp/scripts/service_splitter.py

```diff
@@ -43,6 +43,11 @@
 data = sys.argv
 
 
+# https://stackoverflow.com/questions/1175208/elegant-python-function-to-convert-camelcase-to-snake-case
```

Comment:
My only concern with this is whether or not we strictly obey camel case. Is it possible that we end up feeding something like "GPS" into this, and get g_p_s out? Is it worth it to convert to camel case, or should we just leave it as is?

Reply:
I don't think leaving it as-is is an option.  As noted in the bug report, generated topic services simply don't compile, since the underlying message generation in ROS2 switches file names to camelcase.  It's not just a style issue. That said, I agree that a different function to apply camel case conversion might make sense, particularly if we re-use whatever the ROS2 message generation is using, then it's more likely to be consistent.

Reply:
I changed this in my PR here: https://github.com/swri-robotics/marti_common/pull/604 to use the same logic as the rosidl package does when converting from camel to snake case.